api/index.ts

```diff
@@ -0,0 +1,96 @@
+import axios from 'axios';
+import mitt from 'mitt'
+
+const emitter = mitt();
+
+let token = ''
+let instanceUrl = ''
+
+export function updateToken(key: string) {
+  token = key
+}
+
+export function updateInstanceUrl(url: string) {
+  instanceUrl = url
+}
+
+axios.interceptors.request.use(async (config: any) => {
+    if (token) {
+        config.headers.Authorization =  'Bearer ' + token;
+        config.headers['Content-Type'] = 'application/json';
+    }
+    return config;
+});
+
+axios.interceptors.response.use(function (response) {
+    // Any status code that lie within the range of 2xx cause this function to trigger
+    // Do something with response data
+    return response;
+  }, function (error) {
+    // TODO Handle it in a better way
+    // Currently when the app gets offline, the time between adding a loader and removing it is fractional due to which loader dismiss is called before loader present
+    // which cause loader to run indefinitely
+    // Following gives dismiss loader a delay of 100 microseconds to get both the actions in sync
+    setTimeout(() => emitter.emit("dismissLoader"), 100);
+    if (error.response) {
+        // TODO Handle case for failed queue request
+        const { status } = error.response;
+        if (status == 401) {
+          emitter.emit('unauthorized')
```

Comment:
I think we should define event types like mutations types, so that we have consistency when emit and on/off
https://github.com/hotwax/job-manager/blob/main/src/store/modules/job/mutation-types.ts

Reply:
Defined event type that can use be used for emit and on/off